### PR TITLE
[MM-19266] Manually exit html fullscreen on macOS when in fullscreen

### PR DESCRIPTION
**Pull request by deanwhillier**
Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**
When the Mac Desktop app is in macOS fullscreen, engaging html fullscreen on an inline Youtube video gets stuck in html fullscreen mode. The `Escape` key, the `f` key shortcut and the fullscreen button don't do anything.

This PR listens for the `Escape` and `f` keys from the main process to manually tell the webview in the renderer process to exit html fullscreen if engaged.

This PR doesn't fix the fullscreen button, as due to `contextIsolation` the desktop app can't control the contents of the Webapp.

**Issue link**
https://mattermost.atlassian.net/browse/MM-19226

### CODE REVIEW

File: src/browser/webview/mattermost.js [hunk 1]
```diff
@@ -240,6 +240,13 @@ ipcRenderer.on('user-activity-update', (event, {userIsActive, isSystemEvent}) =>
   window.postMessage({type: 'user-activity-update', message: {userIsActive, manual: isSystemEvent}}, window.location.origin);
 });
 
+// exit fullscreen embedded elements like youtube - https://mattermost.atlassian.net/browse/MM-19226
+ipcRenderer.on('exit-fullscreen', () => {
+  if (document.fullscreenElement) {
```
[the remark this answers is not in the document]
Comment (deanwhillier) [reply src/browser/webview/mattermost.js]: That's what I was going to use, but I switched to this approach after reading here: https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreen. `document.fullscreenElement` is `null` when the document is not in html fullscreen. Can coerce to Boolean if you think that would be better.

File: src/main.js [hunk 2]
```diff
@@ -449,6 +449,13 @@ function handleAppWebContentsCreated(dc, contents) {
   // implemented to temporarily help solve for https://community-daily.mattermost.com/core/pl/b95bi44r4bbnueqzjjxsi46qiw
   contents.on('before-input-event', (event, input) => {
     if (!input.shift && !input.control && !input.alt && !input.meta) {
+      // hacky fix for https://mattermost.atlassian.net/browse/MM-19226
+      if ((input.key === 'Escape' || input.key === 'f') && input.type === 'keyDown') {
```
[the remark this answers is not in the document]
Comment (deanwhillier) [reply src/main.js]: Yes, the 'f' is just the key, no modifiers ... that one worries me as well. This is only happening on mac and the desktop app is already in os fullscreen (not to be confused with html fullscreen), but I agree that there could still be challenges. I'll see about adding the check you suggested!

File: src/browser/webview/mattermost.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Willyfrog) [reply src/browser/webview/mattermost.js]: no, that's perfect then.

File: src/main.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (deanwhillier) [reply src/main.js]: Looks like the best I can do is ensure the fullscreen element is an iframe as any iframe loading a 3rd party site will not be accessible due to cross-domain restrictions.

**Pull request merged**